go/pkg/pass1/set-path.go

```diff
@@ -287,556 +222,497 @@ Partition:
+		zone1 = findZone1(z)
+		partitions2PartitionTags[zone1] =
+			append(partitions2PartitionTags[zone1], z.partition)
+		zone1.partition = z.partition
 	}
+	return partitions2PartitionTags
+}
 
-	// Single unconneted partition does not need to be named.
-	if len(un) == 1 {
-		var partitionName = un[0].partition
+func warnAtNamedSingleUnconnectedPartition(unconnectedPartitions []*zone) {
+	if len(unconnectedPartitions) == 1 {
+		var partitionName = unconnectedPartitions[0].partition
 		if partitionName != "" {
 			warnMsg("Spare partition name for single partition %s: %s.",
-				un[0].name, partitionName)
+				unconnectedPartitions[0].name, partitionName)
 		}
 	}
-	if len(un) > 1 {
-		un = nil
-		for _, unconnectedPartition := range onlyInUn {
-			if unconnectedPartition.ipVxObj.isIPv6() == false {
-				un = append(un, unconnectedPartition)
-			}
-		}
-		if len(un) >= 1 {
-			errMsg("IPv4 topology has unconnected parts:\n%s Use partition attribute, if intended.", nameList(un))
-		}
-	}
-
 }
 
-//##############################################################################
-// Purpose : Include node objects and interfaces of nested loops in the
-//           containing loop; add loop cluster exits; adjust distances of
-//           loop nodes.
-
-func processLoops() { // Meike: statt über router und zonen über pathObj laufen?
-
-	// Check all nodes located inside a cyclic graph.
-	var pathRouters []*router //Meike: managed path routers
-	for _, pathRouter := range allRouters {
-		if pathRouter.managed != "" || pathRouter.semiManaged == true {
-			pathRouters = append(pathRouters, pathRouter)
-		}
-	}
+func errorOnUnnamedUnconnectedPartitions(unconnectedPartitions []*zone,
+	partitions2PartitionTags map[*zone][]string) {
 
-	//Meike: das geht nicht zusammen: einmal zonen, einmal router - 2
-	//schleifen draus machen, später inhalt in funktion auslagern.
-	//für zonen:
-	for _, obj := range zones {
-		myLoop := obj.getLoop()
-		if myLoop == nil {
-			continue
+	if len(unconnectedPartitions) > 1 {
+		var unnamedUnconnectedPartitions []*zone
+		for _, unconnectedPartition := range unconnectedPartitions {
+			if _, exists := partitions2PartitionTags[unconnectedPartition];
+			!exists {
+				unnamedUnconnectedPartitions = append(unnamedUnconnectedPartitions,
+					unconnectedPartition)
+			}
 		}
-
-		// Include sub-loop nodes into containing loop with exit closest to zone1
-		for true {
-			if myLoop.redirect != nil {
-				myLoop = myLoop.redirect
+		if len(unnamedUnconnectedPartitions) >= 1 {
+			var ipVersion string
+			if unconnectedPartitions[0].ipVxObj.isIPv6() {
+				ipVersion = "IPv6"
 			} else {
-				break
+				ipVersion = "IPv4"
+			}
+			var zone1Names []string
+			for _, zone1 := range unnamedUnconnectedPartitions {
+				zone1Names = append(zone1Names, zone1.name)
 			}
+			errMsg("%s topology has unconnected parts:\n - " +
+				"%s\n Use partition attribute, if intended.",
+				ipVersion, strings.Join(zone1Names, "\n - "))
```

Comment:
Hierfür habe ich extra Methoden nameList in error.go definiert:
func (l stringList) nameList() string {
	return " - " + strings.Join(l, "\n - ")
}

Reply:
Das benutze ich jetzt an Stellen, wo die Strings vorher extra eingesammelt werden. Da lege ich dann eine stringList variable für an. Bei   
`errMsg("Several partition names in partition %s:\n - %s",
  zone1.name, strings.Join(partitions2PartitionTags[zone1], "\n - "))`
Ist ja schon eine stringslice da, da lass ich es wie es ist.